### ISSUE TITLE
Using File option doesn't work

### ISSUE THREAD

**Issue by BradyEdgar94** [Zauberfisch/silverstripe-easy-linkfield#2]
Everything works great except the file part. I chose an existing file as the instructions, it looks fine in the backend. However, when I view the frontend it breaks with a `[Notice] Undefined index: type` error.

I debugged it a bit and learned that anytime I add $Link or $File to the frontend, it shows that error. When I also try to use $FileID.URL it's empty.

So something is off with the File part of this that is breaking the frontend.

Here is the error message for reference:
<img width="1376" alt="Screen Shot 2021-06-22 at 8 33 28 PM" src="https://user-images.githubusercontent.com/15221534/123016845-7ddca000-d399-11eb-918a-ee54c2b56135.png">

**Comment by Zauberfisch** [comment]
@BradyEdgar94 I was not able to reproduce your particular error, but I did find a mistake in the FileLink class. I've fixed that now.

Please update to the latest version and in the Template use `$Link` or `$File.URL` (not `$FileID.URL`).  
Please also note, that at the moment it will throw an error if you call a field that doesn't exist. I might change that in the future, but for now, you have to check if the link type is a file and then call `$File.URL`, otherwise you would call `$File` on other Link Types aswell such as InternalLink which does not have a file. So your template should look something like this:

```
<% loop $Buttons.getValue %>
    <%-- Always available Variables: $Link, $AbsoluteLink, $LinkType, $Title, $NewTab --%>
    <%-- And depending on the type: $Page (internal), $PageID (internal), $URL (external), $File (file), $FileID (file), $Email (email), $CountryPrefix (phone), $Number (phone), $PhoneNumber (phone) --%>
    <%-- If you use fields depending on the type, you have to check for the type first, otherwise you will get an error that the field was not found --%>
    <%-- For example <% if $LinkType == 'internal' %>The Link is $Link and the PAGE URLSegment is $Page.URLSegment<% end_if %> --%>
    <% if $LinkType == 'file' %>
        This is File $File.ID ($File.Name) with the URL $Link (or you could use $File.URL)
    <% else %>
        <a href="$Link" <% if $NewTab %>target="_blank"<% end_if %>>$Title</a>
    <% end_if %>
<% end_loop %>
```

**Comment by BradyEdgar94** [comment]
Thanks for looking into this.

So I upgraded but still getting the same issue. I tried uploading a PDF and a PNG but both have the same issue.
<img width="1170" alt="Screen Shot 2021-06-22 at 10 33 18 PM" src="https://user-images.githubusercontent.com/15221534/123026382-111dd180-d3aa-11eb-9b67-0944be75d96a.png">

I then tried the code you provided above but same issues. Its any time I use $Link or $File it throws an error:
<img width="1376" alt="Screen Shot 2021-06-22 at 8 33 28 PM" src="https://user-images.githubusercontent.com/15221534/123026441-2dba0980-d3aa-11eb-8a17-296283112f1b.png">

Not sure why I would be getting this but I can't access that $File object unfortunately.

My SS version is 4.6.1@stable, PHP 7.4. Is there a specific environment I maybe should try?
